Add v3.3 documents #939 (#943)
slackapi/python-slack-sdk: docs-src/scim/index.rst

==============================================
SCIM API Client
==============================================

`SCIM API <https://api.slack.com/scim>`_ is a set of APIs for provisioning and managing user accounts and groups. SCIM is used by Single Sign-On (SSO) services and identity providers to manage people across a variety of tools, including Slack.

> `SCIM (System for Cross-domain Identity Management) <http://www.simplecloud.info/>`_ is supported by myriad services. It behaves slightly differently than other Slack APIs.

Reviewer: ```suggestion
`SCIM (System for Cross-domain Identity Management) <http://www.simplecloud.info/>`_ is supported by a myriad of services. It behaves slightly differently from other Slack APIs.
```

Author: This part is taken from https://api.slack.com/scim - if we change this part, we can change the original one too

Reviewer 2: It also sounded wrong to me, but supposedly it is acceptable to use myriad without of. https://www.dictionary.com/e/how-do-you-use-the-word-myriad/

Author: Thanks, y'all! I've applied Alissa's suggestion to this page. Let's discuss whether we can change the original sentence separately.

Refer to `the API document <https://api.slack.com/scim>`_ for more details.

SCIMClient
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

An OAuth token with `the admin scope <https://api.slack.com/scopes/admin>`_ is required to access the SCIM API.

To fetch provisioned user data, you can use the ``search_users`` method in the client.

.. code-block:: python

    import os
    from slack_sdk.scim import SCIMClient

    client = SCIMClient(token=os.environ["SLACK_ORG_ADMIN_USER_TOKEN"])

    response = client.search_users(
        start_index=1,
        count=100,
        filter="""filter=userName Eq "Carly"""",
    )
    response.users  # List[User]

> Check `the class source code <https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/scim/v1/user.py>`_ to know the structure of the ``user`` in the ``response.users``.

Reviewer: ```suggestion
Check out `the class source code <https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/scim/v1/user.py>`_ to learn more about the structure of the ``user`` in ``response.users``.
```

> Similarly, ``search_groups`` method is available and the ``Group`` object can be `like this <https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/scim/v1/group.py>`_.

Reviewer: ```suggestion
Similarly, the ``search_groups`` method is available and the shape of the ``Group`` object can be `found here <https://github.com/slackapi/python-slack-sdk/blob/main/slack_sdk/scim/v1/group.py>`_.
```

Suggestion made assuming that the phrase 'can be like' was in reference to what the shape was.

.. code-block:: python

    response = client.search_groups(
        start_index=1,
        count=10,
    )
    response.groups  # List[Group]

> For creating, updating, and deleting users/groups,

Reviewer: ```suggestion
For creating, updating, and deleting users/groups:
```

.. code-block:: python

    # POST /Users

>     # Creates a user. Must include the userName attribute and at least one email address.

Reviewer: ```suggestion
    # Creates a user. Must include the user_name argument and at least one email address.
```

>     # You may provide an email address in the userName field,

Reviewer: ```suggestion
    # You may provide an email address as the user_name value,
```
